include/linux/mm_compat_zfs.h

```diff
@@ -0,0 +1,111 @@
+#ifndef	_ZFS_MM_COMPAT_ZFS_H
+#define	_ZFS_MM_COMPAT_ZFS_H
+
+/*
+ * We have various enum members moving between two separate enum types,
+ * and accessed by different functions at various times. Centralise the
+ * insanity.
+ *
+ * < v4.8: all enums in zone_stat_item, via global_page_state()
+ * v4.8: some enums moved to node_stat_item, global_node_page_state() introduced
+ * v4.13: some enums moved from zone_stat_item to node_state_item
+ * v4.14: global_page_state() rename to global_zone_page_state()
+ */
+
+/*
+ * Ensure the config tests are finding one and only one of each enum of interest
+ */
+#if	defined(ZFS_ENUM_NODE_STAT_ITEM_NR_FILE_PAGES) && \
+	defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_FILE_PAGES)
+#error NR_FILE_PAGES found in both node_stat_item and zone_stat_item
+#elif	! defined(ZFS_ENUM_NODE_STAT_ITEM_NR_FILE_PAGES) && \
+	! defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_FILE_PAGES)
+#error NR_FILE_PAGES not found in either node_stat_item or zone_stat_item
+#endif
+
+#if	defined(ZFS_ENUM_NODE_STAT_ITEM_NR_INACTIVE_ANON) && \
+	defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_INACTIVE_ANON)
+#error NR_INACTIVE_ANON found in both node_stat_item and zone_stat_item
+#elif	! defined(ZFS_ENUM_NODE_STAT_ITEM_NR_INACTIVE_ANON) && \
+	! defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_INACTIVE_ANON)
+#error NR_INACTIVE_ANON not found in either node_stat_item or zone_stat_item
+#endif
+
+#if	defined(ZFS_ENUM_NODE_STAT_ITEM_NR_INACTIVE_FILE) && \
+	defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_INACTIVE_FILE)
+#error NR_INACTIVE_FILE found in both node_stat_item and zone_stat_item
+#elif	! defined(ZFS_ENUM_NODE_STAT_ITEM_NR_INACTIVE_FILE) && \
+	! defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_INACTIVE_FILE)
+#error NR_INACTIVE_FILE not found in either node_stat_item or zone_stat_item
+#endif
+
+#if	defined(ZFS_ENUM_NODE_STAT_ITEM_NR_SLAB_RECLAIMABLE) && \
+	defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_SLAB_RECLAIMABLE)
+#error NR_SLAB_RECLAIMABLE found in both node_stat_item and zone_stat_item
+#elif	! defined(ZFS_ENUM_NODE_STAT_ITEM_NR_SLAB_RECLAIMABLE) && \
+	! defined(ZFS_ENUM_ZONE_STAT_ITEM_NR_SLAB_RECLAIMABLE)
+#error NR_SLAB_RECLAIMABLE not found in either node_stat_item or zone_stat_item
+#endif
```

Comment:
Since all of the results from previous configure checks are available at configure time, you could move this sanity check in to a `ZFS_LINUX_TRY_COMPILE` at the end of `ZFS_AC_KERNEL_GLOBAL_PAGE_STATE`.  The upside being you'd detect the issue sooner during `configure` rather than latter with the `make`.

Reply:
Nice. I'll look at that.